pkg/wallet/wallet.go

```diff
@@ -81,6 +81,38 @@ func (w *BaseWallet) CreateSigningKey() (string, error) {
 	return base58Pub, nil
 }
 
+// ConvertToEncryptionKey converts an ed25519 keypair present in the wallet,
+// persists the resulting keypair, and returns the result public key.
+func (w *BaseWallet) ConvertToEncryptionKey(key string) (string, error) {
+	signPub := base58.Decode(key)
+	encPub, err := cryptoutil.PublicEd25519toCurve25519(signPub)
+	if err != nil {
+		return "", err
+	}
+	encPubB58 := base58.Encode(encPub)
+
+	kp, err := w.getKey(key)
+	if err != nil {
+		return "", err
+	}
+	encPriv, err := cryptoutil.SecretEd25519toCurve25519(kp.Priv)
+	if err != nil {
+		return "", err
+	}
+	kpEnc := cryptoutil.KeyPair{Priv: encPriv, Pub: encPub}
+	err = w.persistKey(encPubB58, &kpEnc)
+	if err != nil {
+		return "", err
+	}
+
+	return encPubB58, nil
+}
+
+// GetBox returns a Box which provides crypto box encryption which can use this wallet's secret keys
+func (w *BaseWallet) GetBox() *Box {
```

Comment:
I’m not sure that this method “gets” something.

Reply:
I would suggest reversing the injection.

I.e, have a NewBox method at the package level that takes a wallet.